src/main/java/mate/academy/rickandmorty/service/CharactersApiClient.java

```diff
@@ -0,0 +1,35 @@
+package mate.academy.rickandmorty.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.external.ApiResponseDto;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class CharactersApiClient {
+    private final ObjectMapper objectMapper;
+    private final HttpClient httpClient = HttpClient.newHttpClient();
+
+    public ApiResponseDto getAllCharacterFromApi(String url) {

```

Comment:
better save this url somewhere here
```suggestion
    public ApiResponseDto getAllCharacterFromApi() {
```